### PR TITLE
HLT menu development for `13_0_X` (4/N) [`13_1_X`]

### PR DESCRIPTION
**Pull request by missirol**
#### PR description:

Development of HLT menus for `CMSSW_13_0_X`.

This is the PR for the `master` branch, which will then be backported to `13_0_X` (the latter is the release cycle currently used for Run-3 HLT-menu development).

This PR includes the integration of the JIRA tickets in [1], and has the following implications.

 - The GRun menu in this PR is compatible with the L1T menu [`L1Menu_Collisions2023_v1_0_0`](https://htmlpreview.github.io/?https://github.com/cms-l1-dpg/L1MenuRun3/blob/57fcce7ecf26366084813755f769f47be58bbf5f/development/L1Menu_Collisions2023_v1_0_0/L1Menu_Collisions2023_v1_0_0.html) [[tag]](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/tags/L1Menu_Collisions2023_v1_0_0_xml) (see [CMSLITDPG-1102](https://its.cern.ch/jira/browse/CMSLITDPG-1102)).

 - This 2023 L1T menu will be included by AlCa in the 2023/24 MC and Run-3 'data relval' GTs (https://github.com/cms-sw/cmssw/pull/41005), while the 2022 MC GTs continue to use the last L1T menu of 2022. To comply with this, the alias `(HLT:)@relval2023` is introduced to run the GRun menu in the 2023/24 MC and Run-3 data RelVals, and the alias `(HLT:)@relval2022` is made to point to the frozen 2022 HLT menu. To pick up the appropriate HLT menu in all relevant wfs, changes in `Configuration/PyReleaseValidation` were needed.

[1]
  - [CMSHLT-2678](https://its.cern.ch/jira/browse/CMSHLT-2678): [HCAL] removed `L1_SingleMuOpen` seed from `HLT_HcalPhiSym_v` trigger
  - [CMSHLT-2687](https://its.cern.ch/jira/browse/CMSHLT-2687): [BPH] update of `Tau3Mu` triggers (2023 L1T seeds)
  - [CMSHLT-2688](https://its.cern.ch/jira/browse/CMSHLT-2688): [BPH] removed `L1_DoubleMu3er2p0_SQ_OS_dR_Max1p4` seed from BPH dimuon triggers
  - [CMSHLT-2689](https://its.cern.ch/jira/browse/CMSHLT-2689): [JME] use L1T-Jet-FWD2p5 seeds in `PFJetFwd` and `DiPFJetAve` triggers (2023 L1T seeds)
  - [CMSHLT-2690](https://its.cern.ch/jira/browse/CMSHLT-2690): [BTV] update of `BTagMu*DiJet` triggers (2023 L1T seeds)
  - [CMSHLT-2699](https://its.cern.ch/jira/browse/CMSHLT-2699): [HIG] added backup L1T VBFDiJetIsoTau seed (2023 L1T seed)

#### PR validation:

TSG tests.

#### If this PR is a backport, please specify the original PR and why you need to backport that PR. If this PR will be backported, please specify to which release cycle the backport is meant for:

`CMSSW_13_0_X`

### CODE REVIEW

**Comment by missirol** [comment]
test parameters:
  - pull_requests = #41005
  - workflows = 139.004,140.003,140.023,140.043,140.063,12634.0,12634.99,12834.0,13034.0,13034.99
  - enable = gpu

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by missirol** [comment]
+hlt

 - The changes in the outputs of PR tests are in line with the updates to the HLT menus used in the 20(22|23|24) MC RelVals and Run-3 data RelVals.

 - The Phase-2 wfs are unaffected by this PR (and [#41005](https://github.com/cms-sw/cmssw/pull/41005)). Still, they show spurious differences in the `JetMET/{Jet,MET}Validation` (e.g. [wf 20900.0](https://tinyurl.com/2p4cvesg)). This is reminiscent of [#39754](https://github.com/cms-sw/cmssw/pull/39754), and most unlikely unrelated to this PR or [#41005](https://github.com/cms-sw/cmssw/pull/41005) (in fact, such differences are not seen in the tests of [#41005 (comment)](https://github.com/cms-sw/cmssw/pull/41005#issuecomment-1462896237), which included this PR).

**Comment by missirol** [comment]
@cms-sw/pdmv-l2 @cms-sw/upgrade-l2

Could you please prioritise the review of this PR (and its backport)?

The latest GT updates by AlCa (#41005) require this HLT PR to be integrated.

File: Configuration/PyReleaseValidation/python/relval_standard.py [hunk 1]
```diff
@@ -443,72 +443,72 @@
 workflows[136.9] = ['', ['RunDoubleMuon2016C', 'RAWRECOTE16', 'RAWRECOLHECLEANTE16', 'EMBEDHADTE16', 'EMBEDMINIAOD16']]
 
 ### run 2021 collisions ###
-workflows[139.001] = ['RunMinimumBias2021',['RunMinimumBias2021','HLTDR3_2022','RECODR3_reHLT_MinBiasOffline','HARVESTD2021MB_reHLT']]
-workflows[139.002] = ['',['RunZeroBias2021','HLTDR3_2022','RECODR3_reHLT_ZBOffline','HARVESTD2021ZB_reHLT']]
-workflows[139.003] = ['',['RunHLTPhy2021','HLTDR3_2022','RECODR3_reHLT_HLTPhysics_Offline','HARVESTD2021HLTPhy_reHLT']]
-workflows[139.004] = ['',['RunNoBPTX2021','HLTDR3_2022','RECODR3_reHLT_AlCaTkCosmics_Offline','HARVESTDR3_reHLT']]
+workflows[139.001] = ['RunMinimumBias2021',['RunMinimumBias2021','HLTDR3_2023','RECODR3_reHLT_MinBiasOffline','HARVESTD2021MB_reHLT']]
```
Comment (srimanob) [inline Configuration/PyReleaseValidation/python/relval_standard.py:446]: Do we want to reHLT of previous year data with new HLT menu? Compare to Run-2 reHLT in the same file, we do reHLT using the menu of the same year data taking.

Comment (missirol) [reply Configuration/PyReleaseValidation/python/relval_standard.py:446]: Yes (at least for now). The reason is that, in 13_1_X and 13_0_X, the HLT GT used by these workflows (`run3_hlt_relval`) uses the 2023 L1T menu, so these wfs require this HLT menu. In general, I think we should have Run-3 data workflows that exercise the latest Run-3 HLT menu (as done here).

**Comment by srimanob** [comment]
+Upgrade

**Comment by kskovpen** [comment]
+pdmv

**Comment by missirol** [comment]
@cms-sw/orp-l2

This PR is ready for your review.

This PR and #41005 need to be integrated in the same IB.

**Comment by rappoccio** [comment]
+1

Also merging #41005

**Pull request merged**